File: frontend/src/api/modules/llm-settings.ts
import request from '../request'

export interface LLMProviderDTO {
  id: number
  name: string
  type: string
  base_url: string
  api_key?: string
  model: string
  max_tokens: number
  is_default: boolean
  preset_id?: string
  protocol?: string
  created_at: string
  updated_at: string
}

export interface LLMPresetModel {
  id: string
  display_name: string
}

export interface LLMPresetDTO {
  id: string
  display_name: string
  category: 'coding_plan' | 'direct' | 'local'
  type: string
  base_url: string
  anthropic_url?: string
  supports_anthropic: boolean
  models: LLMPresetModel[]
  default_model: string
  max_tokens: number
  icon: string
  region: 'cn' | 'global' | 'local'
  requires_key: boolean
  is_coding_plan: boolean
  coding_plan_price?: string
  warning?: string
  subscribe_url?: string
}

export interface CodeReviewGlobalSettingsDTO {
  enabled: boolean
  auto_review_on_mr: boolean
  block_on_high: boolean
  max_files: number
  max_diff_lines: number
}

export function listLLMProviders() {
  return request.get<unknown, LLMProviderDTO[]>('/settings/llm-providers')
}

export function fetchLLMPresets(category?: string) {
  const params = category ? { category } : {}
  return request.get<unknown, LLMPresetDTO[]>('/settings/llm-providers/presets', { params })
}

export function getLLMProvider(id: number) {
  return request.get<unknown, LLMProviderDTO>(`/settings/llm-providers/${id}`)
}

export function createLLMProvider(data: Partial<LLMProviderDTO>) {
  return request.post<unknown, LLMProviderDTO>('/settings/llm-providers', data)
}

export function updateLLMProvider(id: number, data: Partial<LLMProviderDTO>) {
  return request.put<unknown, LLMProviderDTO>(`/settings/llm-providers/${id}`, data)
}

export function deleteLLMProvider(id: number) {
  return request.delete<unknown, any>(`/settings/llm-providers/${id}`)
}

export function setDefaultLLMProvider(id: number) {
  return request.put<unknown, any>(`/settings/llm-providers/${id}/default`)
}

export function testLLMProvider(id: number) {
  return request.post<unknown, { status: string; message: string }>(`/settings/llm-providers/${id}/test`)
}

export function fetchOllamaModels(baseUrl?: string) {
  const params = baseUrl ? { base_url: baseUrl } : {}
  return request.get<unknown, string[]>('/settings/ollama-models', { params })
}

export function getCodeReviewSettings() {
  return request.get<unknown, CodeReviewGlobalSettingsDTO>('/settings/code-review')
}

export function updateCodeReviewSettings(data: CodeReviewGlobalSettingsDTO) {
  return request.put<unknown, CodeReviewGlobalSettingsDTO>('/settings/code-review', data)
}

File: biz/model/api/llm_settings.go
package api

import "github.com/yi-nology/git-manage-service/biz/model/po"

type LLMProviderDTO struct {
	ID             uint   `json:"id"`
	Name           string `json:"name"`
	Type           string `json:"type"`
	BaseURL        string `json:"base_url"`
	APIKey         string `json:"api_key,omitempty"`
	Model          string `json:"model"`
	MaxTokens      int    `json:"max_tokens"`
	IsDefault      bool   `json:"is_default"`
	IsEmbedding    bool   `json:"is_embedding"`
	EmbeddingModel string `json:"embedding_model"`
	PresetID       string `json:"preset_id,omitempty"`
	Protocol       string `json:"protocol,omitempty"`
	CreatedAt      string `json:"created_at"`
	UpdatedAt      string `json:"updated_at"`
}

func NewLLMProviderDTO(p po.LLMProvider) LLMProviderDTO {
	return LLMProviderDTO{
		ID:             p.ID,
		Name:           p.Name,
		Type:           p.Type,
		BaseURL:        p.BaseURL,
		APIKey:         p.APIKey,
		Model:          p.AIModel,
		MaxTokens:      p.MaxTokens,
		IsDefault:      p.IsDefault,
		IsEmbedding:    p.IsEmbedding,
		EmbeddingModel: p.EmbeddingModel,
		PresetID:       p.PresetID,
		Protocol:       p.Protocol,
		CreatedAt:      p.CreatedAt.Format("2006-01-02T15:04:05Z"),
		UpdatedAt:      p.UpdatedAt.Format("2006-01-02T15:04:05Z"),
	}
}

type CodeReviewGlobalSettingsDTO struct {
	Enabled        bool `json:"enabled"`
	AutoReviewOnMR bool `json:"auto_review_on_mr"`
	BlockOnHigh    bool `json:"block_on_high"`
	MaxFiles       int  `json:"max_files"`
	MaxDiffLines   int  `json:"max_diff_lines"`
	RAGEnabled     bool `json:"rag_enabled"`
}
